.github/workflows/pr-test.yml

```diff
@@ -0,0 +1,52 @@
+name: test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    steps:
+       # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+      
+      - name: create checksum file
+        run: find . -type f -name "*.gradle*" -o -name "gradle-wrapper*" | xargs shasum | sort > checksum.txt && cat checksum.txt
+
+      - name: Cache packages
+        id: cache-packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle
+          key: gradle-packages-${{ runner.os }}-${{ github.job }}-${{ hashFiles('**/checksum.txt') }}
+          restore-keys: |
+            gradle-packages-${{ runner.os }}-${{ github.job }}
+            gradle-packages-${{ runner.os }}
+
+      - name: Unit test
+        run: ./gradlew jacocoTestReport
```

Comment:
nit: is there a way of displaying test results in GHA - CCI has a way of uploading test results and I'd guess there's some equivalent for GHA. At a minimum, uploading the junit html report as a test artifact, if nothing first class for tests.

Reply:
yup. we can upload artifacts.